src/v/cluster/partition.cc

```diff
@@ -202,8 +204,25 @@ ss::future<std::error_code> partition::prefix_truncate(
           _raft->ntp());
         co_return make_error_code(cluster::errc::feature_disabled);
     }
-    co_return co_await _log_eviction_stm->truncate(
-      truncation_offset, deadline, _as);
+    vlog(
+      clusterlog.info,
+      "Truncating {} to redpanda offset {} kafka offset {}",
+      _raft->ntp(),
+      truncation_offset,
+      kafka_truncation_offset);
+    if (truncation_offset != model::offset{}) {
```

Comment:
What is truncation_offset is == model::offset{} ? Shouldn't we just return OOR? FWIW i see this is later removed

Reply:
It's {} if we couldn't translate it, because the kafka offset fell below the local log.